zerver/migrations/0458_remove_invalid_characters_from_system_group_name.py

```diff
@@ -0,0 +1,38 @@
+# Generated by Django 4.2.2 on 2023-07-03 06:39
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def remove_invalid_characters_from_system_group_name(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    # Remove "@" character from name of role-based system groups.
+    UserGroup = apps.get_model("zerver", "UserGroup")
+
+    groups_to_update = []
+    for group in UserGroup.objects.filter(is_system_group=True):
+        if group.name[0] != "@":
+            # We skip the group if its name has already been updated.
+            continue
+
+        old_group_name = group.name
+        group.name = old_group_name[1:]
+        groups_to_update.append(group)
+
+    UserGroup.objects.bulk_update(groups_to_update, ["name"])
```

Comment:
This migration I think should do a second pass where it renames any non-system groups that violate the naming rules established here. I checked Zulip Cloud and unsurprisingly nobody has named groups with names starting with things like `stream:`; group names starting with `@` are extremely rare and don't have a group in the same organization with `@` removed present.

So I'd propose that if a non-system-group has a name that violates the rules defined here, we should just rename it to remove the leading `@` (or more generally, whatever the problematic prefix is), and if that fails, rename it to `group_{id}`, where ID is its integer ID; I expect that fallback code will not run in practice but the safe fallback seems helpful to ensure the migration runs.